### PR TITLE
Shift execution context of rspace opreations

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Title says all

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/CORE-1028

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] If this PR adds a new feature, this PR includes tests related to this feature.
- [x] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the [developer wiki](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain) for instructions.
- [ ] Your GitHub account is also an account with [Travis CI](https://travis-ci.org). Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala [hunk 1]
```diff
@@ -28,13 +31,42 @@ object RSpace {
         create(LockFreeInMemoryStore.create(ctx.trieStore, branch), branch)
     }
 
-  private[rspace] def create[F[_], C, P, E, A, R, K](store: IStore[C, P, A, K], branch: Branch)(
+  def createInMemory[F[_], C, P, E, A, R, K](
+      trieStore: ITrieStore[InMemTransaction[history.State[Blake2b256Hash, GNAT[C, P, A, K]]], Blake2b256Hash, GNAT[
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: maybe:
`type InMemTrieStore[C, P, A, K] = ITrieStore[InMemTransaction[history.State[Blake2b256Hash, GNAT[C, P, A, K]]], Blake2b256Hash, GNAT[C, P, A, K]]`?

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala [hunk 2]
```diff
@@ -28,13 +31,42 @@ object RSpace {
...
+
+    val mainStore = InMemoryStore
+      .create[InMemTransaction[history.State[Blake2b256Hash, GNAT[C, P, A, K]]], C, P, A, K](
+        trieStore,
+        branch
+      )
+    create(mainStore, branch)
+  }
+
+  def create[F[_], C, P, E, A, R, K](store: IStore[C, P, A, K], branch: Branch)(
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: just a thought: `createInMemory` and `create` don't align well naming wise.

File: rspace/src/main/scala/coop/rchain/rspace/spaces/FineGrainedRSpace.scala [hunk 3]
```diff
@@ -161,137 +169,139 @@ class FineGrainedRSpace[F[_], C, P, E, A, R, K] private[rspace] (
...
-        span.mark("before-produce-lock")
-        produceLock(channel) {
-          span.mark("produce-lock-acquired")
-          //TODO fix double join fetch
-          val groupedChannels: Seq[Seq[C]] = store.withTxn(store.createTxnRead()) { txn =>
-            store.getJoin(txn, channel)
-          }
-          span.mark("grouped-channels")
-          logger.debug(s"""|produce: searching for matching continuations
+    contextShift.evalOn(scheduler) {
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/spaces/FineGrainedRSpace.scala]: maybe a similar refactor to replay (lockedConsume, lockedProduce)?

Comment (lukasz-golebiewski) [reply rspace/src/main/scala/coop/rchain/rspace/spaces/FineGrainedRSpace.scala]: I don't see a point at the moment

**Pull request merged**